src/test/ui/async-await/async-fn-nonsend.stderr

```diff
@@ -45,9 +45,9 @@ LL |     assert_send(non_sync_with_method_call());
    = note: required because of the requirements on the impl of `std::marker::Send` for `&mut dyn std::fmt::Write`
    = note: required because it appears within the type `std::fmt::Formatter<'_>`
    = note: required because of the requirements on the impl of `std::marker::Send` for `&mut std::fmt::Formatter<'_>`
-   = note: required because it appears within the type `for<'r, 's> {&'r mut std::fmt::Formatter<'s>, bool, bool, fn() -> impl std::future::Future {fut}, impl std::future::Future, impl std::future::Future, ()}`
-   = note: required because it appears within the type `[static generator@$DIR/async-fn-nonsend.rs:39:38: 45:2 for<'r, 's> {&'r mut std::fmt::Formatter<'s>, bool, bool, fn() -> impl std::future::Future {fut}, impl std::future::Future, impl std::future::Future, ()}]`
-   = note: required because it appears within the type `std::future::GenFuture<[static generator@$DIR/async-fn-nonsend.rs:39:38: 45:2 for<'r, 's> {&'r mut std::fmt::Formatter<'s>, bool, bool, fn() -> impl std::future::Future {fut}, impl std::future::Future, impl std::future::Future, ()}]>`
+   = note: required because it appears within the type `for<'r, 's> {&'r mut std::fmt::Formatter<'s>, bool, bool, fn(impl std::future::Future) -> <impl std::future::Future as std::future::IntoFuture>::Future {<impl std::future::Future as std::future::IntoFuture>::into_future}, fn() -> impl std::future::Future {fut}, impl std::future::Future, impl std::future::Future, impl std::future::Future, ()}`
+   = note: required because it appears within the type `[static generator@$DIR/async-fn-nonsend.rs:39:38: 45:2 for<'r, 's> {&'r mut std::fmt::Formatter<'s>, bool, bool, fn(impl std::future::Future) -> <impl std::future::Future as std::future::IntoFuture>::Future {<impl std::future::Future as std::future::IntoFuture>::into_future}, fn() -> impl std::future::Future {fut}, impl std::future::Future, impl std::future::Future, impl std::future::Future, ()}]`
```

Comment:
My opinion is that these diagnostics, while worse than the originals, were already pretty bad, so I don't think it's a problem to land them as is.

Reply:
@estebank This does remind me however of the issue that we need to shorten paths in diagnostics for paths in scope... when will that be fixed? :D

Reply:
I think that further progress along the lines of https://github.com/rust-lang/rust/pull/65456 might help here. I tried rebasing this PR on top of it, but when running this test I get the following with no span

    error[E0433]: failed to resolve: could not find `IntoFuture` in `future`